src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Fx2Nl/Fx2NlLanguageServerOperationHandler.cs

```diff
@@ -52,17 +52,17 @@ private Task PreHandleFx2NlAsync(LanguageServerOperationContext operationContext
             (scope) =>
             {
                 _preHandleCheckResult = scope?.Check(_fx2NlRequestParams.Expression) ?? throw new NullReferenceException("Check result was not found for Fx2NL operation");
-                _fx2NlParameters = GetFx2NlHints(scope, operationContext);
+                _fx2NlParameters = GetFx2NlParameters(scope, operationContext);
             }, cancellationToken);
         }
 
         /// <summary>
-        /// Gets the Fx2Nl hints.
+        /// Gets the Fx2Nl hints and range.
         /// </summary>
         /// <param name="scope"> PowerFx Scope.</param>
         /// <param name="operationContext">Language Server Operation Context.</param>
-        /// <returns>Fx2Nl hints.</returns>
-        private Fx2NLParameters GetFx2NlHints(IPowerFxScope scope,  LanguageServerOperationContext operationContext)
+        /// <returns>Fx2Nl parameters including hints and range.</returns>
+        private Fx2NLParameters GetFx2NlParameters(IPowerFxScope scope,  LanguageServerOperationContext operationContext)

```

Comment:
wait why are you not automatically transferring the range to fx2nlparameters from customfx2nl params? that should be done here so not all hosts have to remember transferring range from custom fx2nl params to fx2nlparameters

Reply:
Ah okay let me take a look at it.